src/TLE.ts

```diff
@@ -874,65 +874,78 @@ export class Parser {
         assert(1 <= stringValue.length, "TLE strings must be at least 1 character.");
         assert(Utilities.isQuoteCharacter(stringValue[0]), "The first character in a TLE string must be a quote character.");
 
-        let leftSquareBracketToken: Token = null;
+        const leftSquareBracketToken: Token = null;
         let expression: Value = null;
-        let rightSquareBracketToken: Token = null;
-        let errors: language.Issue[] = [];
+        const rightSquareBracketToken: Token = null;
+        const errors: language.Issue[] = [];
+        let result: { stringValue: string; leftSquareBracketToken: Token; expression: Value; rightSquareBracketToken: Token; errors: language.Issue[] } = { stringValue, leftSquareBracketToken, expression, rightSquareBracketToken, errors };
+
+        let closingBracketLocation = stringValue.length - 1;
+        for (closingBracketLocation; closingBracketLocation > 0; closingBracketLocation--) {
```

Comment:
What's the intention of this code?  Is it to handle the case where the string is unterminated?  Again, getting too complicated.

Let's get rid of this and turn the condition around:

    if (stringValue.startsWith('"[') && stringValue.endsWith(']"') && !stringValue.startsWith('"[[')) {
       // This is considered an expression
…



            expression = new StringValue(Token.createQuotedString(0, stringValue)); 


            stringValue[1] !== "[" // Does not start with '[' 




            || stringValue.substr(1, 2) === "[[" // Starts with '[[' 




            || (stringValue[1] === '[' && stringValue[closingBracketLocation] !== ']' && stringValue.indexOf(']') >= 0) // Starts with '[' but doesn't end with ']' 




        ) { 




            // This is considered a plain string